### PR TITLE
feat(components/pages): add page header sub-components

### PR DESCRIPTION
**Pull request by Blackbaud-ErikaMcVey**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1512/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/forms-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/1512/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1512/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1512/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1512/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1512/playground/)

File: libs/components/pages/src/lib/modules/page/page.component.ts [hunk 1]
```diff
@@ -31,10 +31,9 @@ const LAYOUT_CLASS_DEFAULT = `${LAYOUT_CLASS_PREFIX}${LAYOUT_DEFAULT}`;
 export class SkyPageComponent implements OnInit, OnDestroy {
   /**
    * The page layout that corresponds with the top-level component type
-   * used on the page. For laying out custom content, use `auto` to allow
-   * the page contents to expand beyond the bottom of the browser window
-   * or `fit` to constrain the page contents to the available viewport.
-   * @default "auto"
+   * used on the page.`fit` to constrains the page contents to the available viewport.
```
Comment (Blackbaud-SteveBrush) [inline libs/components/pages/src/lib/modules/page/page.component.ts]: Seems like there might be a typo in this sentence.

File: libs/components/pages/src/lib/modules/page-header/page-header.component.spec.ts [hunk 2]
```diff
@@ -2,25 +2,87 @@ import { TestBed } from '@angular/core/testing';
...
+    fixture.detectChanges();
+
+    const smallAvatar = fixture.nativeElement.querySelector(
+      '.sky-avatar-wrapper-size-small'
+    );
+
+    expect(smallAvatar).toBeVisible();
+  });
+
+  it('should render page a page header alert', () => {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/pages/src/lib/modules/page-header/page-header.component.spec.ts]: `page a page`

File: libs/components/avatar/src/lib/modules/avatar/avatar.component.ts [hunk 3]
```diff
@@ -80,32 +84,28 @@ export class SkyAvatarComponent {
...
-  }
+  public size: SkyAvatarSize | undefined;
 
   /**
    * Emits a `SkyFileItem` object when the image is updated.
    */
   @Output()
   public avatarChanged = new EventEmitter<SkyFileItem>();
 
+  public sizeDefault: Observable<SkyAvatarSize> | undefined;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/avatar/src/lib/modules/avatar/avatar.component.ts]: Possible to make this `protected`?

File: libs/components/core/src/lib/modules/default-input-provider/default-input-provider.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import { Observable, ReplaySubject } from 'rxjs';
+
+/**
+ * @internal
+ * An API to provide default Angular component input values to child components.
+ */
+export class SkyDefaultInputProvider {
+  #props: Record<string, Record<string, ReplaySubject<unknown>>> = {};
+
+  public setValue(
```
Comment (Blackbaud-SteveBrush) [inline libs/components/core/src/lib/modules/default-input-provider/default-input-provider.ts]: Would be nice to have a generic applied to the `setValue` method as well.
```
public setValue<T>(componentName: string, inputName: string, value: T)
```

File: libs/components/core/src/lib/modules/default-input-provider/default-input-provider.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  #getSubject(
+    componentName: string,
+    inputName: string
+  ): ReplaySubject<unknown> {
+    const inputSubject = this.#props[componentName]?.[inputName];
+
+    if (!inputSubject) {
+      const componentSubjects = this.#props[componentName] || {};
```
Comment (Blackbaud-SteveBrush) [inline libs/components/core/src/lib/modules/default-input-provider/default-input-provider.ts]: You're reading from `this.#props[componentName]` in three different places (two of which have null checks). You could assign it once to simplify:
```
#getSubject(
  componentName: string,
  inputName: string
): ReplaySubject<unknown> {
  const componentSubjects = this.#props[componentName] || {};
  const inputSubject = componentSubjects[inputName];

  if (!inputSubject) {
    componentSubjects[inputName] = new ReplaySubject(1);
    this.#props[componentName] = componentSubjects;
  }

  return componentSubjects[inputName];
}
```

File: libs/components/pages/src/lib/modules/page/page.component.ts [hunk 6]
```diff
@@ -31,10 +31,9 @@ const LAYOUT_CLASS_DEFAULT = `${LAYOUT_CLASS_PREFIX}${LAYOUT_DEFAULT}`;
...
+   * `none` is intended for custom content and does not apply any spacing or constraints.
+   * @default "none"
```
Comment (Blackbaud-SteveBrush) [inline libs/components/pages/src/lib/modules/page/page.component.ts:37]: We're changing the default value in a non-breaking change?

File: libs/components/avatar/src/lib/modules/avatar/avatar.component.ts [hunk 7]
```diff
@@ -80,32 +84,28 @@ export class SkyAvatarComponent {
...
 
-  #_size: SkyAvatarSize = 'large';
-
   #_src: SkyAvatarSrc | undefined;
 
   #errorService: SkyErrorModalService;
   #fileSizePipe: SkyFileSizePipe;
   #resourcesService: SkyLibResourcesService;
 
+  #defaults = inject(SkyDefaultInputProvider, { optional: true });
```
Comment (Blackbaud-SteveBrush) [inline libs/components/avatar/src/lib/modules/avatar/avatar.component.ts]: When I read `#defaults` I assume this property holds all defaults for the component. Could we instead name it `#defaultInputProvider` so it's more clear in the code what it is and how to use it?

File: libs/components/pages/src/lib/modules/page/page.component.ts (hunk 6 above)
Comment (Blackbaud-ErikaMcVey) [reply libs/components/pages/src/lib/modules/page/page.component.ts:37]: @Blackbaud-SteveBrush it has always been the default--it was just a mistake in the docs

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**